alma/lib/Utils/CartDataHelper.php

```diff
@@ -0,0 +1,368 @@
+<?php
+/**
+ * 2018-2022 Alma SAS
+ *
+ * THE MIT LICENSE
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+ * documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+ * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+ * Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+ * WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ *
+ * @author    Alma SAS <contact@getalma.eu>
+ * @copyright 2018-2022 Alma SAS
+ * @license   https://opensource.org/licenses/MIT The MIT License
+ */
+
+namespace Alma\PrestaShop\Utils;
+
+use Address;
+use Alma\PrestaShop\Model\CartData;
+use Alma\PrestaShop\Model\ShippingData;
+use Cart;
+use Context;
+use Country;
+use Customer;
+use State;
+use Tools;
+use Validate;
+
+if (!defined('_PS_VERSION_')) {
+    exit;
+}
+
+/**
+ * Class CartDataHelper.
+ *
+ * Cart Data
+ */
+class CartDataHelper
+{
+    public function __construct(
+        Cart $cart,
+        Context $context,
+        array $feePlans
+    )
+    {
+        $this->cart = $cart;
+        $this->context = $context;
+        $this->feePlans = $feePlans;
```

Comment:
```suggestion
        $this->feePlan = $feePlan;
```

it seems to be a single selected fee-plan actually

Reply:
No we did a foreach in getEligibilityQueries